### ISSUE TITLE
How can I limit pdfminer to process stuff in the cropbox or mediabox?

### ISSUE THREAD

**Issue by Vricken** [pdfminer/pdfminer.six#786]
It is not a bug I was just wondering this:
How can I limit pdfminer to process stuff in the cropbox or mediabox?

**Comment by KunalGehlot** [comment]
I believe you want to selectively process the items within a defined range of coordinates. Is that right? 

You could mine the whole pdf and then selectively pick the elements which lie in your defined coordinates by their x1, y1, x2 and y2 values.

**Comment by pietermarsman** [comment]
To completely ignore elements from the PDF document you could extend or completely replace the `PDFPageInterpreter`. This class receives all the "commands" from the PDF. E.g. displaying text in a certain position or drawing lines. You should use the [composable api](https://pdfminersix.readthedocs.io/en/latest/tutorial/composable.html) to do this. 

Another option is to replace the `PDFDevice`. E.g. `TextConverter` is the `PDFDevice` that is used to print text to the console. The `PDFDevice` is responsible for the layout analysis. You cannot change the `PDFDevice` to prevent reading and processing parts of the PDF page, but you can use it to exclude certain components from the layout analysis (which is often the heaviest part of the processing). 

In summary, a low-level but more performant approach is to extend or replace the `PDFPageInterpreter`. An higher-level and less performant approach is to extend or replace the `PDFDevice`.
